canonical-web-and-design/react-components
# Generate prop types

## Done

- Added a babel plugin to generate prop types: [babel-plugin-typescript-to-proptypes](https://github.com/milesj/babel-plugin-typescript-to-proptypes).
- Removed prop types from all components.

## QA

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Open Accordion.stories.mdx.
- Add a arg with the wrong type to the default story e.g. `onExpandedChange: false,`
- Start storybook.
- Open the accordion story and there should be a console error.

## Fixes

Fixes: #201.
